### PR TITLE
Update Sweep ref docs

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_limits.py [hunk 1]
```diff
@@ -83,6 +75,16 @@ class SweepJobLimits(JobLimits):
     :type timeout: int
     :param trial_timeout: Sweep Job Trial timeout value in seconds.
     :type trial_timeout: int
+
+    .. admonition:: Example:
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_limits.py:79]: Is admonition some sort of docstring keyword? Edit - nevermind, found out below.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py [hunk 2]
```diff
@@ -107,6 +140,13 @@ def __init__(
         delay_evaluation: int = 0,
         evaluation_interval: int = 1,
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py:141]: Should these (and other numbers throughout the file) also be optional-wrapped?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py [hunk 3]
```diff
@@ -107,6 +140,13 @@ def __init__(
...
     ) -> None:
+        """Defines an early termination policy based on running averages of the primary metric of all runs.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py:143]: Should "running averages" be singular here? It sounds like it's a single average of one metric, assuming I'm reading this correctly.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py [hunk 4]
```diff
@@ -15,7 +15,13 @@
 
 
 class SamplingAlgorithm(ABC, RestTranslatableMixin):
+    """Base class for sampling algorithms.
+
+    You should not instantiate this class directly. Instead, use of its subclasses.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py:20]: nit: Missing word "one":
```suggestion
    You should not instantiate this class directly. Instead, use one of its subclasses.
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py [hunk 5]
```diff
@@ -39,16 +45,24 @@ def _from_rest_object(cls, obj: RestSamplingAlgorithm) -> "SamplingAlgorithm":
...
-        based random sampling.
-    :type logbase: Union[float, str]
     :param rule: The specific type of random algorithm. Possible values include: "random",
         "sobol".
     :type rule: str
-    :param seed: An integer to use as the seed for random number generation.
+    :param seed: The seed for random number generation.
     :type seed: int
+    :param logbase: A positive number or e in string format to be used as base for log
+        based random sampling.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py:54]: Nit: quotes around "e" for clarity, and added "the":
```suggestion
    :param logbase: A positive number or "e" in string format to be used as the base for
        log-based random sampling.
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py [hunk 6]
```diff
@@ -58,6 +72,17 @@ def __init__(
         seed: int = None,
         logbase: Union[float, str] = None,
     ) -> None:
+        """Random Sampling Algorithm.
+
+        :param rule: The specific type of random algorithm. Possible values include: "random",
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py:77]: Are there other potential values, or is this list comprehensive? If the latter, should we reword it to "Acceptable values are:"?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py [hunk 7]
```diff
@@ -58,6 +72,17 @@ def __init__(
...
+            "sobol".
+        :type rule: str
+        :param seed: The seed for random number generation.
+        :type seed: int
+        :param logbase: A positive number or e in string format to be used as base for log
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sampling_algorithm.py:82]: nit: similar to other logbase.
```suggestion
    :param logbase: A positive number or "e" in string format to be used as the base for
        log-based random sampling.
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/search_space.py [hunk 8]
```diff
@@ -14,7 +14,18 @@
...
+    """Base class for sweep distribution configuration.
+
+    :param type: Type of distribution.
+    :type type: str
+    """
+
+    def __init__(self, *, type: Optional[str] = None) -> None:  # pylint: disable=redefined-builtin
+        """Sweep distribution configuration.
+
+        :param type: Type of distribution.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/search_space.py:26]: Does this have a small list of acceptable values that's worth listing?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py [hunk 9]
```diff
@@ -148,7 +158,57 @@ def __init__(
...
+    ) -> None:
+        """Sweep job for hyperparameter tuning.
+
+        :param name: Name of the job.
+        :type name: str
+        :param display_name: Display name of the job.
+        :type display_name: str
+        :param description: Description of the job.
+        :type description: str
+        :param tags: Tag dictionary. Tags can be added, removed, and updated.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py:170]: The second sentence implies that this can be used for somewhat complex behavior. Does the tags argument simply fully replace existing tags when non-empty, or do something more complex?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py [hunk 10]
```diff
@@ -148,7 +158,57 @@ def __init__(
...
+        :type tags: dict[str, str]
+        :param properties: The asset property dictionary.
+        :type properties: dict[str, str]
+        :param experiment_name:  Name of the experiment the job will be created under, if None is provided,
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py:174]: Wording nit: sentence break:
```suggestion
        :param experiment_name:  Name of the experiment the job will be created under. If None is provided,
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py [hunk 11]
```diff
@@ -148,7 +158,57 @@ def __init__(
...
+            job will be created under experiment 'Default'.
+        :type experiment_name: str
+        :param identity: Identity that training job will use while running on compute.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py:177]: wording nit: added "the"
```suggestion
        :param identity: Identity that the training job will use while running on compute.
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py [hunk 12]
```diff
@@ -148,7 +158,57 @@ def __init__(
...
+            hyperparameter and the value is the parameter expression.
+        :type search_space: Dict
+        :param objective: Metric to optimize for.
+        :type objective: Objective
+        :param compute: The compute target the job runs on.
+        :type compute: str
+        :param trial: The job configuration for each trial. Each trial will be provided with a different combination
+            of hyperparameter values that the system samples from the search_space.
+        :type trial: Union[~azure.ai.ml.entities.CommandJob, ~azure.ai.ml.entities.CommandComponent]
+        :param early_termination: The early termination policy to use.A trial job is canceled
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py:199]: Nit: Space after period:
```suggestion
        :param early_termination: The early termination policy to use. A trial job is canceled
```

File: sdk/ml/azure-ai-ml/documentation_guidelines.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+6. **Coverage**: Every public object must have a docstring. Docstrings are encouraged, but not required for private objects. 
+7. **Demonstration**: Every docstring should include an up-to-date code snippet that demonstrates how to use the product effectively. 
+
+
+##### Accuracy 
+
+- No broken links are allowed in reference documentation, and all links must be location/language neutral (e.g. no /en-us/ in links)
+- Docstring details should be aligned with type hints 
+- Any code snippets should be defined in the samples directory in this folder so that they are tested during every CI run
+- Constructors (`def __init__`) should return None, per [PEP 484 standards](https://peps.python.org/pep-0484/#the-meaning-of-annotations)
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/documentation_guidelines.md:21]: Nit: periods
```suggestion
- No broken links are allowed in reference documentation, and all links must be location/language neutral (e.g. no /en-us/ in links).
- Docstring details should be aligned with type hints.
- Any code snippets should be defined in the samples directory in this folder so that they are tested during every CI run.
- Constructors (`def __init__`) should return None, per [PEP 484 standards](https://peps.python.org/pep-0484/#the-meaning-of-annotations).
```

File: sdk/ml/azure-ai-ml/documentation_guidelines.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+ 
+##### Completeness 
+
+- Be sure to warn of any edge cases or note special circumstances, if applicable. This can be done using an admonition in the docstring. In addition to the “tip” class we use for examples, Sphinx provides classes for other types of messages that you should feel free to use if appropriate:
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/documentation_guidelines.md:26]: Nit: Taking this chance to explicitly state what sphinx is for people who are unfamiliar with it:
```suggestion
- Be sure to warn of any edge cases or note special circumstances, if applicable. This can be done using an admonition in the docstring. In addition to the “tip” class we use for examples, Sphinx (the python docs generator that we use) provides classes for other types of messages that you should feel free to use if appropriate: 
```

Edit: Nevermind, you do that below.

File: sdk/ml/azure-ai-ml/documentation_guidelines.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+       This is an important admonition. 
+    
+      
+    
+    .. admonition:: Information 
+    
+       :class: info 
+    
+       This is an informational admonition. 
+    ```
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/documentation_guidelines.md:58]: Double checking - is the extra spacing between every line intended here?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py (hunk 3 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/early_termination_policy.py:143]: This is the summary that MFE provided for this class, but I agree with your reading. Sounds like singular fits best. I'll make the change.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py (hunk 9 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/sweep_job.py:170]: The tags param comes from the base Resource class that almost all entities inherit. If you assign a dictionary as the tags param, it will override the existing one. If you try to update it like a dictionary, it will update. I'll find some language to fix here, but it will be repo-wide since this param and description is used everywhere.

File: sdk/ml/azure-ai-ml/documentation_guidelines.md (hunk 15 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/documentation_guidelines.md:58]: It's irrelevant - it wouldn't affect the formatting ultimately. I can make it single-spaced.

File: sdk/ml/azure-ai-ml/documentation_guidelines.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+##### Relevance
+
+- If you deprecate an object, make sure it’s documented in the reference docs. 
+- If you change how an object should be used, ensure that all code snippets in the samples directory in this folder are updated accordingly.
+
+ 
+
+##### Coverage 
+
+- AzureML v2 is working to get 95% documentation coverage of the entire codebase. ALL new code that you write or existing code that you edit must include a docstring following all of the guidelines. This will be enforced via the interrogate Python package in CI in the future. In the meantime, you can use [interrogate](https://interrogate.readthedocs.io/en/latest/) locally to keep yourself honest.
```
[the remark this answers is not in the document]
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/documentation_guidelines.md:90]: The CLI command instruction is included at the official website link where they'd need to go to install it, but I think it's fair to suggest we should have those things in this doc. I'll add them.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/search_space.py (hunk 8 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/sweep/search_space.py:26]: Oh, I forgot about the uninstantiability... in that case it depends on how often you suspect that we'll need to be doing dev work using this class. If we're going to be call this often, then being able to just look at this to see the options all list in one place would be nice. But if this is likely to be relatively untouched code most of the time, then we can keep it concise.

For now I'll leave it as resolved. This isn't a choice that needs to be made now.

**Review by MilesHolland — APPROVED** [review]
